provides.py

```diff
@@ -0,0 +1,23 @@
+from charmhelpers.core import hookenv
+from charms.reactive import when, when_not
+from charms.reactive import set_flag, clear_flag
+from charms.reactive import Endpoint
+
+
+class MountProvides(Endpoint):
```

Comment:
This side needs a way to get the list of applications requesting mounts.  Something like:

```python
@property
def applications(self):
    return [rel.application_name for rel in self.relations]
```